modin/experimental/core/execution/dask/implementations/pandas_on_dask/io/io.py

```diff
@@ -0,0 +1,126 @@
+# Licensed to Modin Development Team under one or more contributor license agreements.
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+"""
+Module houses experimental IO classes and parser functions needed for these classes.
+
+Any function or class can be considered experimental API if it is not strictly replicating existent
+Query Compiler API, even if it is only extending the API.
+"""
+
+import warnings
+
+import pandas
+
+from modin.core.storage_formats.pandas.parsers import (
+    PandasCSVGlobParser,
+    PandasPickleExperimentalParser,
+    CustomTextExperimentalParser,
+)
+from modin.core.storage_formats.pandas.query_compiler import PandasQueryCompiler
+from modin.core.execution.dask.implementations.pandas_on_dask.io import PandasOnDaskIO
+from modin.core.io import (
+    CSVGlobDispatcher,
+    PickleExperimentalDispatcher,
+    CustomTextExperimentalDispatcher,
+)
+from modin.experimental.core.io import SQLExperimentalDispatcher
+
+from modin.core.execution.dask.implementations.pandas_on_dask.dataframe import (
+    PandasOnDaskDataframe,
+)
+from modin.core.execution.dask.implementations.pandas_on_dask.partitioning import (
+    PandasOnDaskDataframePartition,
+)
+from modin.core.execution.dask.common import DaskWrapper
+
+
+class ExperimentalPandasOnDaskIO(PandasOnDaskIO):
+    """
+    Class for handling experimental IO functionality with pandas storage format and Dask engine.
+
+    ``ExperimentalPandasOnDaskIO`` inherits some util functions and unmodified IO functions
+    from ``PandasOnDaskIO`` class.
+    """
+
+    build_args = dict(
+        frame_partition_cls=PandasOnDaskDataframePartition,
+        query_compiler_cls=PandasQueryCompiler,
+        frame_cls=PandasOnDaskDataframe,
+        base_io=PandasOnDaskIO,
+    )
+    read_csv_glob = type(
+        "", (DaskWrapper, PandasCSVGlobParser, CSVGlobDispatcher), build_args
+    )._read
```

Comment:
I wonder if it's worth doing something like
```python
def make_read(*classes):
    return type("", (DaskWrapper, *classes), build_args)._read
read_csv_glob = make_read(PandasCSVGlobParser, CSVGlobDispatcher)
# other methods
del make_read # to not pollute class namespace
```
?

Reply:
I really like this idea.

Reply:
implemented